packages/dd-trace/src/appsec/index.js

```diff
@@ -262,6 +259,21 @@ function onResponseSetHeader ({ res, abortController }) {
   }
 }
 
+function onRequestProcessParams ({ req, res, abortController, params }) {
+  const rootSpan = web.root(req)
+  if (!rootSpan) return
+
+  if (!params || typeof params !== 'object' || !Object.keys(params).length) return
```

Comment:
is this the only place where we have Object.keys check ? why is that

Reply:
In the other site - `TaintTrackingPlugin`- , the key length check is done (indirectly) later on, in the subscriber handler (`taintObject` method).

Her, this is done to avoid an unnecessary call to the WAF.